src/metatrain/utils/additive/zbl.py

```diff
@@ -0,0 +1,307 @@
+                for nl, system in zip(neighbor_lists, systems)
+            ]
+        )
+        zj = torch.concatenate(
+            [
+                system.types[nl.samples.column("second_atom")]
+                for nl, system in zip(neighbor_lists, systems)
+            ]
+        )
+
+        # Find the interatomic distances
+        rij = torch.concatenate(
+            [torch.sqrt(torch.sum(nl.values**2, dim=(1, 2))) for nl in neighbor_lists]
+        )
+
+        # Find the ZBL energies
+        e_zbl = self.get_pairwise_zbl(zi, zj, rij)
+
+        # Sum over edges to get node energies
+        indices_for_sum_list = []
+        sum = 0
+        for system, nl in zip(systems, neighbor_lists):
+            indices_for_sum_list.append(nl.samples.column("first_atom") + sum)
+            sum += system.positions.shape[0]
+
+        e_zbl_nodes = torch.zeros(sum, dtype=e_zbl.dtype, device=e_zbl.device)
+        e_zbl_nodes.index_add_(0, torch.cat(indices_for_sum_list), e_zbl)
+
+        device = systems[0].positions.device
+
+        # Set the outputs as the ZBL energies
+        targets_out: Dict[str, TensorMap] = {}
+        for target_key, target in outputs.items():
+            if target_key.startswith("mtt::aux::"):
+                continue
+            sample_values: List[List[int]] = []
+
+            for i_system, system in enumerate(systems):
+                sample_values += [[i_system, i_atom] for i_atom in range(len(system))]
+
+            block = TensorBlock(
+                values=e_zbl_nodes.reshape(-1, 1),
+                samples=Labels(
+                    ["system", "atom"], torch.tensor(sample_values, device=device)
+                ),
+                components=[],
+                properties=Labels(
+                    names=["energy"], values=torch.tensor([[0]], device=device)
+                ),
+            )
+
+            targets_out[target_key] = TensorMap(
+                keys=Labels(names=["_"], values=torch.tensor([[0]], device=device)),
+                blocks=[block],
+            )
+
+            # apply selected_atoms to the composition if needed
+            if selected_atoms is not None:
+                targets_out[target_key] = metatensor.torch.slice(
+                    targets_out[target_key], "samples", selected_atoms
+                )
+
+            if not target.per_atom:
+                targets_out[target_key] = metatensor.torch.sum_over_samples(
+                    targets_out[target_key], sample_names="atom"
+                )
+
+        return targets_out
+
+    def get_pairwise_zbl(self, zi, zj, rij):
+        """
+        Ziegler-Biersack-Littmark (ZBL) potential.
+
+        Inputs are the atomic numbers (zi, zj) of the two atoms of interest
+        and their distance rij.
+        """
+        # set cutoff from covalent radii of the elements
+        rc = (
+            self.covalent_radii[self.species_to_index[zi]]
+            + self.covalent_radii[self.species_to_index[zj]]
```

Comment:
I think you are indexing the wrong element here. You should decrement the index by one. `covalent_radii` is an array representing the radii and starts counting from 0 rather 1. For example hydrogen has number atomic number 1 which is represented in `covalent_radii` by index 0.

Reply:
We digged a bit more and the current indexing is correct